flow/layers/container_layer.cc

```diff
@@ -10,6 +10,102 @@ namespace flutter {
 
 ContainerLayer::ContainerLayer() {}
 
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+
+void ContainerLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  auto old_container = static_cast<const ContainerLayer*>(old_layer);
+  DiffContext::AutoSubtreeRestore subtree(context);
```

Comment:
Nit: it seems that whenever a `DiffContext* context` needs to do something (e.g., `SetLayerPaintRegion`, `PushTransform`, ...), we have to call `DiffContext::AutoSubtreeRestore subtree(context)` first in the scope. I guess it would be bad if some future call sites forget to call `DiffContext::AutoSubtreeRestore subtree(context);`. To avoid it, shall we move those methods into `DiffContext::AutoSubtreeRestore` so one cannot directly call `context-> SetLayerPaintRegion(...)`. Instead, one is forced to call
```
DiffContext::AutoSubtreeRestore subtree(context);
subtree.SetLayerPaintRegion(...);
```

Reply:
I was actually thinking about doing the modifications indirectly like this but I don't think I've seen RAII wrapper used for mutations anywhere else in the project. I'll give it another look.